services-camel/src/test/java/edu/unc/lib/cdr/thumbnail/ThumbnailRouterTest.java

```diff
@@ -0,0 +1,227 @@
+		String body = createBody();
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put(IDENTIFIER, "container");
+		
+		template.sendBodyAndHeaders("direct-vm:createThumbnail", body, headers);
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testResourceTypeFilter() throws Exception {
+		getMockEndpoint("mock:fcrepo:{{fcrepo.baseUri}}").expectedMessageCount(0);
+		
+		createContext();
+		
+		String body = createBody();
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put(RESOURCE_TYPE, createResource( "http://bad.info/definitions/v9/repository#Fake" ).getURI());
+		
+		template.sendBodyAndHeaders("direct-vm:createThumbnail", body, headers);
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testRouteMulticastSuccess() throws Exception {
+		context.getRouteDefinition("IsImage").adviceWith((ModelCamelContext) context, new AdviceWithRouteBuilder() {
+			@Override
+			public void configure() throws Exception {
+				replaceFromWith("direct:start");
+				mockEndpointsAndSkip("*");
+			}
+		});
+		context.start();
+		getMockEndpoint("mock:direct:small.thumbnail").expectedMessageCount(1);
+		getMockEndpoint("mock:direct:large.thumbnail").expectedMessageCount(1);
+		template.sendBodyAndHeaders("", createEvent(fileID, eventTypes));
+		
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testRouteMulticastFilter() throws Exception {
+		context.getRouteDefinition("IsImage").adviceWith((ModelCamelContext) context, new AdviceWithRouteBuilder() {
+			@Override
+			public void configure() throws Exception {
+				replaceFromWith("direct:start");
+				mockEndpointsAndSkip("*");
+			}
+		});
+		context.start();
+		getMockEndpoint("mock:direct:small.thumbnail").expectedMessageCount(0);
+		getMockEndpoint("mock:direct:large.thumbnail").expectedMessageCount(0);
+		
+		Map<String, Object> headers = createEvent(fileID, eventTypes);
+		headers.put("MimeType", "plain/text");
+		
+		template.sendBodyAndHeaders("", headers);
+		
+		assertMockEndpointsSatisfied();
+	}
+	
+	private String createBody() throws IOException {
+		PID pid = PIDs.get(UUID.randomUUID().toString());
+		Model model = ModelFactory.createDefaultModel();
+		Resource resc = model.createResource(pid.getRepositoryPath());
+		resc.addProperty(hasMimeType, "application/octet-stream");
+		resc.addProperty(hasMessageDigest, "123456789");
+		
+		// Serialize the object as a string so that the processor can receive it
+		String body = null;
+		try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+			body = new String(bos.toByteArray(), "UTF-8");
+		}
+		
+		return body;
+	}
+	
+	private void createContext() throws Exception {
+		context.getRouteDefinition("CdrServiceEnhancements").adviceWith((ModelCamelContext) context, new AdviceWithRouteBuilder() {
+			@Override
+			public void configure() throws Exception {
+				mockEndpointsAndSkip("fcrepo*");
+				weaveById("simpleBinaryMetadataProcessor");
```

Comment:
Tests pass without this line and I can't find any references to something with that id, so I think we can safely remove this line